drivers/include/l3gxxxx.h

```diff
@@ -0,0 +1,1928 @@
+/**
+ * @brief   Read one sample of raw sensor data as 16 bit two's complements
+ *
+ * Function works only in bypass mode and fails in FIFO modes. In FIFO modes,
+ * function *l3gxxxx_get_raw_data_fifo* has to be used instead to get data.
+ *
+ * @param[in]   dev     device descriptor of L3Gxxxx sensor
+ * @param       raw     raw data vector
+ *
+ * @retval  L3GXXXX_OK      on success
+ * @retval  L3GXXXX_ERROR_* negative error code, see #l3gxxxx_error_codes_t
+ */
+int l3gxxxx_read_raw(const l3gxxxx_t *dev, l3gxxxx_raw_data_t *raw);
+
+/** @} */
+
+#if IS_USED(MODULE_L3GXXXX_FIFO) || DOXYGEN
+
+/**
+ * @name    FIFO data handling
+ * @{
+ */
+
+#if IS_USED(MODULE_L3GXXXX_CONFIG) || DOXYGEN
+/**
+ * @brief   Set FIFO mode
+ *
+ * The use of the integrated FIFO of the sensor allows to reduce the
+ * interaction events of the MCU with the sensor and thus to save power.
+ * Data samples can be fetched as a burst if there is a sufficient number of
+ * data samples in the FIFO. This number is defined as threshold or
+ * watermark and an interrupt can be triggered when number of data samples
+ * in the FIFO becomes greater than or equal to this threshold.
+ *
+ * Furthermore, different FIFO modes can be defined which determine the
+ * behavior in the case the FIFO is full, see #l3gxxxx_fifo_mode_t.
+ *
+ * This function is available only if modules `l3gxxxx_fifo`
+ * and `l3gxxxx_config` are used.
+ *
+ * @param[in]   dev     device descriptor of L3Gxxxx sensor
+ * @param[in]   mode    FIFO mode
+ * @param[in]   thresh  FIFO watermark (ignored in bypass mode)
+ *
+ * @retval  L3GXXXX_OK      on success
+ * @retval  L3GXXXX_ERROR_* negative error code, see #l3gxxxx_error_codes_t
+ */
+int l3gxxxx_set_fifo_mode(l3gxxxx_t *dev,
+                          l3gxxxx_fifo_mode_t mode, uint8_t thresh);
+
+#endif /* IS_USED(MODULE_L3GXXXX_CONFIG) || DOXYGEN */
+
+/**
+ * @brief   Get all samples of sensor data stored in the FIFO (unit dps)
+ *
+ * In bypass mode, it returns only one sensor data sample.
+ *
+ * This function is available only if module `l3gxxxx_fifo` is used.
+ *
+ * @param[in]   dev     device descriptor of L3Gxxxx sensor
+ * @param[out]  data    array of 32 data structures
+ *
+ * @return  number of data samples read on success, or negative error code,
+ *          see #l3gxxxx_error_codes_t
+ */
+int l3gxxxx_read_fifo(const l3gxxxx_t *dev,
+                      l3gxxxx_data_fifo_t data);
+
+/**
+ * @brief   Get all samples of raw sensor data stored in the FIFO
+ *
+ * In bypass mode, it returns only one raw data sample.
+ *
+ * This function is available only if module `l3gxxxx_fifo` is used.
+ *
+ * @param[in]   dev     device descriptor of L3Gxxxx sensor
+ * @param[out]  raw     array of 32 raw data structures
+ *
+ * @return  number of data samples read on success, or negative error code,
+ *          see #l3gxxxx_error_codes_t
+ */
+int l3gxxxx_read_raw_fifo(const l3gxxxx_t *dev,
+                          l3gxxxx_raw_data_fifo_t raw);
```

Comment:
Does it make sense to provide this as a public function?

Reply:
Hm, when I wrote the driver for the CCS811 gas sensor for another project some years ago, someone insisted on having a function to read the raw data from sensor, although calculating the physical data from raw sensor data is quite difficult for such sensors. Since that time I also expose the function to read the raw data from the sensor.

I can hide it if you think it is better.